### ISSUE TITLE
Tiles do not display to left of meridian

### ISSUE THREAD

**Issue by scadergit** [openlayers/openlayers#14219]
**Describe the bug**
The short version is the extent does not wrap in the view.

Our geoserver provides tiles and the extent of the tiles. The extent is queried from the geoserver then applied to a tile wms layer with the geoserver layer source. The extent is the entire map:

```(-180.00000000000003,-85.05999999999999,180.00000000000003,85.05999999999999)```

When plotted, this looks like:
<img src="https://user-images.githubusercontent.com/17500494/196529637-f7f9e2ca-d4f8-4e82-bba9-8a87bdb5d9e4.png" width="300">
Here is the example with the missing tiles, which can be found if you pan all the way to the right:
<img src="https://user-images.githubusercontent.com/17500494/196534858-dd393efd-79d1-4959-a1b6-ce1b3568c0cc.png" width="300">
<img src="https://user-images.githubusercontent.com/17500494/196535152-b99f1aad-daeb-472a-83ef-2ccd72c8cad6.png" width="300">
If I exclude the extent, it displays as expected:
<img src="https://user-images.githubusercontent.com/17500494/196535595-00701fa9-b778-4435-a762-defda0542bdb.png" width="300">
There seem to have been properties in previous versions that addressed this issue, but have been removed from code since.

**Expected behavior**
The tile wms layer extent should wrap in the view, allowing tiles to the left of the meridian to display.

**Comment by ahocevar** [comment]
> There seem to have been properties in previous versions that addressed this issue, but have been removed from code since.

Which ones are you referring to?

If your layer spans the whole world, you do not need to set a layer extent.

**Comment by scadergit** [comment]
Yeah, obvious answer is obvious, but in some cases it won't be full map, so I asked anyways. =)

I found one called wrapDateLine, but not sure if that fully correlated to this issue.

**Comment by scadergit** [comment]
@ahocevar so would you say this is expected behavior?

**Comment by ahocevar** [comment]
Yes, because extent means to constrain the view to that extent.

**Comment by scadergit** [comment]
True, but the extent wraps in the view, but the tiles to do not. So the missing tiles are within the definition of the extent.

**Comment by ahocevar** [comment]
Do you have `wrapX` set to `true` on the source?

**Comment by scadergit** [comment]
According to the docs that defaults to yes, but I set it anyways to true and it didn't allow the tiles to wrap.

**Comment by ahocevar** [comment]
Can you maybe provide a working example, e.g. by modifying one of the official examples on https://openlayers.org/en/latest/examples/ using the 'Edit' button, that shows the problem? I think we can consider this behavior a bug, so now we need an example to derive a test case from, and then a fix.

**Comment by scadergit** [comment]
Yes, just wanted to make sure it was a possible bug. I'll draft something next week.
